### PR TITLE
Add Support for Orch 20.10

### PR DESCRIPTION
**Pull request by rajivml**
* AIFBR-6814
* Add Orchestrator 20.10 Support and make it backward compatible with Orch 20.4

### CODE REVIEW

File: orchestrator/orchestratorAutomation.ps1 [hunk 1]
```diff
@@ -35,26 +39,50 @@ Param (
...
+if( $config -notmatch '\\$' ){
+    $config += '\'
+}
 
-    if($domainBasedAccess.Length -gt 0){
-        $domainBasedAccess = $domainBasedAccess.ToString()
-    } else {
-        $domainBasedAccess = "false"
-    }
+$dll_config = $config + 'UiPath.Orchestrator.dll.config'
```
Comment (mohitseth-nitrr) [inline orchestrator/orchestratorAutomation.ps1:54]: Can we check if this file exists and if it has valid permissions ?

File: orchestrator/orchestratorAutomation.ps1 [hunk 2]
```diff
@@ -35,26 +39,50 @@ Param (
...
 
-echo "Path to web.config: "$config
+#Fetching Orchestrator version
+if(Test-Path $dll_config){
+    $orchestrator_version = [System.Diagnostics.FileVersionInfo]::GetVersionInfo($config + 'UiPath.Orchestrator.web.dll').FileVersion
+    echo "Orchestrator version : $orchestrator_version"
+}
+
+if(Test-Path $dll_config){
```
Comment (mohitseth-nitrr) [inline orchestrator/orchestratorAutomation.ps1:62]: Can we check if this file exists and if it has valid permissions ?

File: orchestrator/orchestratorAutomation.ps1 [hunk 3]
```diff
@@ -35,26 +39,50 @@ Param (
...
+
+if($domainBasedAccess.Length -gt 0){
+    $domainBasedAccess = $domainBasedAccess.ToString()
+} else {
+    $domainBasedAccess = "false"
+}
+
+echo "Path to Web config: "$config
 
 Copy-Item $config -Destination ("$config.original."+(Get-Date -Format "MMddyyyy.HH.mm.ss"))
```
Comment (mohitseth-nitrr) [inline orchestrator/orchestratorAutomation.ps1:88]: i have doubts about this condition as even with domainBasedAccess ... we host aifabric on 31390 only ... please recheck this

**Review by mohitseth-nitrr — CHANGES_REQUESTED** [review]
changes needed

File: orchestrator/orchestratorAutomation.ps1 (hunk 2 above)
Comment (rajivml) [reply orchestrator/orchestratorAutomation.ps1:62]: that is what am checking with Test-Path i.e if the file exists and the script is supposed to be edited in powershell administrator mode so permission check is not required

File: orchestrator/orchestratorAutomation.ps1 (hunk 1 above)
Comment (rajivml) [reply orchestrator/orchestratorAutomation.ps1:54]: script is already running using powershell administrator mode, so permission check is not required

Comment (mohitseth-nitrr) [reply orchestrator/orchestratorAutomation.ps1:54]: actually we faced this issue with one of the customer ... he entered wrong path for config and then the script failed ... it took sometime to figure out that path was wrong. 
Lets check for each file and path existence to avoid such cases in future.

File: orchestrator/orchestratorAutomation.ps1 (hunk 2 above)
Comment (mohitseth-nitrr) [reply orchestrator/orchestratorAutomation.ps1:62]: sure ... sounds good 👍

**Review by mohitseth-nitrr — CHANGES_REQUESTED** [review]
changes needed

File: orchestrator/orchestratorAutomation.ps1 (hunk 1 above)
Comment (rajivml) [reply orchestrator/orchestratorAutomation.ps1:54]: file check has been added

File: orchestrator/orchestratorAutomation.ps1 (hunk 3 above)
Comment (rajivml) [reply orchestrator/orchestratorAutomation.ps1:94]: nope, with domainBasedAccess i.e AKS case, we use DNS name to access ai-app and there is no need of specifying 31390 port over there

Comment (mohitseth-nitrr) [reply orchestrator/orchestratorAutomation.ps1:94]: can we rephrase it to portLess access in that case as for one box installation even with domainBasedAccess, we need port number ?

Comment (mohitseth-nitrr) [reply orchestrator/orchestratorAutomation.ps1:94]: or something better as it was confusing for customer at that point

Comment (rajivml) [reply orchestrator/orchestratorAutomation.ps1:94]: ok, renamed it

File: orchestrator/orchestratorAutomation.ps1 [hunk 4]
```diff
@@ -16,7 +16,11 @@
     ./orchestratorAutomation.ps1 -aifip ww.xx.yy.zz -orcname orchestrator.uipath.com
 
     If ai-app is accessed via domain instead of IP:PORT combo, then enable domainBasedAccess to true
-    .\orchestratorAutomation.ps1 -aifip "aif-sahil-aks.westeurope.cloudapp.azure.com" -orcname "aifabricdevorch.northeurope.cloudapp.azure.com" -domainBasedAccess "true"
+    .\orchestratorAutomation.ps1 -aifip "aif-sahil-aks.westeurope.cloudapp.azure.com" -orcname "aifabricdevorch.northeurope.cloudapp.azure.com" -portlessAccess "true"
```
Comment (vivekjhauipath) [inline orchestrator/orchestratorAutomation.ps1:19]: shouldn't we have if else here? Are we expecting users to comment one of these lines before running?

Comment (vivekjhauipath) [reply orchestrator/orchestratorAutomation.ps1:19]: btw, are we taking a stance that we will only support orch 20.10? If not, then let's have existing script too.

File: orchestrator/orchestratorAutomation.ps1 [hunk 5]
```diff
@@ -30,36 +34,66 @@ Param (
...
-        $config = "C:\Program Files (x86)\UiPath\Orchestrator\web.config"
-    }
+Import-Module 'WebAdministration' 
+    
 
-    if(!$aifport){   
-        $aifport = "31390"
-    }
+if(!$config){   
+    $config = "C:\Program Files (x86)\UiPath\Orchestrator"
```
Comment (vivekjhauipath) [inline orchestrator/orchestratorAutomation.ps1:46]: why is config folder location?

Comment (vivekjhauipath) [reply orchestrator/orchestratorAutomation.ps1:46]: let's call it baseFolder

**Review by vivekjhauipath — COMMENTED** [review]
Need some clarifications

File: orchestrator/orchestratorAutomation.ps1 (hunk 4 above)
Comment (rajivml) [reply orchestrator/orchestratorAutomation.ps1:19]: @vivekjhauipath we are supporting both 20.4 and 20.10 config edit... the script is backward compatible with 20.4 as well ... and regarding your first comment , it's just documentation on various ways on how can we run that script

**Review by mohitseth-nitrr — APPROVED** [review]
LGTM

File: orchestrator/orchestratorAutomation.ps1 (hunk 5 above)
Comment (rajivml) [reply orchestrator/orchestratorAutomation.ps1:46]: it's just a cosmetic change vivek , if we modify it, we have to change the documentation as well...

**Review by Mdirfan6 — APPROVED** [review]
lgtm
